### PR TITLE
Request retry mechanism

### PR DESCRIPTION
**Pull request by superDross**
This allows users to configure how they want their requests to be retried.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -66,16 +66,24 @@ Command line usage:
     wlc download
     wlc upload
 
-Configuration is stored in ``~/.config/weblate``:
+Configuration is stored in ``~/.config/weblate``. The key/values (``retries``,
```
Comment (kamilgrymuza) [inline README.rst:69]: Is it possible to set those via environment variables? In docker environments it's sometimes tricky to inject a file?

File: wlc/__init__.py [hunk 2]
```diff
@@ -147,8 +186,7 @@ def post(self, path, **kwargs):
 
     def _post_factory(self, prefix, path, kwargs):
         """Wrapper for posting objects."""
-        resp = self.post("/".join((prefix, path, "")), **kwargs)
-        return resp
+        return self.post("/".join((prefix, path, "")), **kwargs)
```
Comment (kamilgrymuza) [inline wlc/__init__.py:189]: Nice 👍

**Review by kamilgrymuza — APPROVED** [review]
Looks good to me @superDross. The only thing I would consider adding is the ability to provide the configuration via the environment.

File: README.rst (hunk 1 above)
Comment (superDross) [reply README.rst:69]: The existing pattern for configuration is to parse them directly as a parameter to the `wlc.Weblate` class or by adding them to a config file.

I am currently using env vars to configure our `wlc.Weblate` object in Dragoman by parsing them directly to the class upon initilisation:

```python
Weblate(
    key=os.environ.get('WEBLATE_KEY', None),
    url=os.environ.get('WEBLATE_URL', None),
    retries=os.environ.get('RETRIES', 3),
    status_forcelist=os.environ.get(
        'STATUS_FORCELIST',
       [429, 500, 502, 503, 504]
    )
    method_whitelist=os.environ.get(
        "METHOD_WHITELIST",
        ["HEAD", "TRACE", "DELETE", "PUT", "OPTIONS", "GET", "POST"]
    )
    backoff_factor=os.environ.get("BACKOFF_FACTOR", 0.2),
)
```
We can therefore parse the request retry configurations in dragoman-ui and dragoman apps `APP_SETTINGS_YAML` and then export them to the environment when running `app.py`. This is how I have been configuring it with Dragoman when testing my weblate integration changes.

I have allowed this to continue by adding parameters for the request configuration to the [`wlc.Weblate` class](https://github.com/superDross/wlc/blob/a7d931a119cc8da565e65d07bbb42db116b5c54b/wlc/__init__.py#L66).

I don't think we need to add the ability to configure `wlc` directly with env vars.

Comment (kamilgrymuza) [reply README.rst:69]: Makes sense David - thanks for a detailed explanation.

**Review by kamilgrymuza — APPROVED** [review]
Looks good to me 👍

**Pull request merged**